lib/core/src/system/web/dom/html/css3d_renderer.rs

```diff
@@ -0,0 +1,269 @@
+use web_sys::HtmlElement;
+use basegl_system_web::get_element_by_id;
+
+
+// ===================
+// === Js Bindings ===
+// ===================
+
+mod js {
+    use super::*;
+    #[wasm_bindgen(module = "/src/system/web/dom/html/snippets.js")]
+    extern "C" {
+        #[allow(unsafe_code)]
+        pub fn setup_perspective(dom: &JsValue, znear: &JsValue);
+
+        #[allow(unsafe_code)]
+        pub fn setup_camera_orthographic(dom:&JsValue, matrix_array:&JsValue);
+
+        #[allow(unsafe_code)]
+        pub fn setup_camera_perspective
+        ( dom          : &JsValue
+        , near         : &JsValue
+        , matrix_array : &JsValue
+        );
+    }
+}
+
+/// Inverts Matrix Y coordinates.
+/// It's equivalent to scaling by (1.0, -1.0, 1.0).
+pub fn invert_y(mut m: Matrix4<f32>) -> Matrix4<f32> {
+    // Negating the second column to invert Y.
+    m.row_part_mut(1, 4).iter_mut().for_each(|a| *a = -*a);
+    m
+}
+
+#[allow(unsafe_code)]
+fn setup_camera_perspective
+(dom:&JsValue, near:f32, matrix:&Matrix4<f32>) { // Views to WASM memory are only valid as long the backing buffer isn't
+    // resized. Check documentation of IntoFloat32ArrayView trait for more
+    // details.
+    unsafe {
+        let matrix_array = matrix.js_buffer_view();
+        js::setup_camera_perspective(
+            &dom,
+            &near.into(),
+            &matrix_array
+        )
+    }
+}
+
+#[allow(unsafe_code)]
+fn setup_camera_orthographic(dom:&JsValue, matrix:&Matrix4<f32>) {
+    // Views to WASM memory are only valid as long the backing buffer isn't
+    // resized. Check documentation of IntoFloat32ArrayView trait for more
+    // details.
+    unsafe {
+        let matrix_array = matrix.js_buffer_view();
+        js::setup_camera_orthographic(&dom, &matrix_array)
+    }
+}
+
+
+
+// =========================
+// === Css3dRendererData ===
+// =========================
+
+#[derive(Debug)]
+pub struct Css3dRendererData {
+    pub front_dom    : HtmlElement,
+    pub back_dom     : HtmlElement,
+    pub front_camera : HtmlElement,
+    pub back_camera  : HtmlElement
+}
+
+impl Css3dRendererData {
+    pub fn new
+    ( front_dom:HtmlElement
+    , back_dom:HtmlElement
+    , front_camera:HtmlElement
+    , back_camera:HtmlElement) -> Self {
+        Self {front_dom,back_dom,front_camera,back_camera}
+    }
+}
+
+impl Css3dRendererData {
+    fn set_dimensions(&self, dimensions:Vector2<f32>) {
+        let width  = format!("{}px", dimensions.x);
+        let height = format!("{}px", dimensions.y);
+        let doms   = vec![&self.front_dom,&self.back_dom,&self.front_camera,&self.back_camera];
+        for dom in doms {
+            dom.set_property_or_panic("width" , &width);
+            dom.set_property_or_panic("height", &height);
+        }
+    }
+}
+
+// ====================
+// === Css3dRenderer ===
+// ====================
+
+/// A renderer for `Css3dObject`s.
+#[derive(Clone,Debug)]
+pub struct Css3dRenderer {
```

Comment:
This is a very important struct. It needs better docs. Please describe what it does, what are its responsibilities. When they are fired, etc.

Reply:
What about this?
```rust
/// `Css3dRenderer` is a renderer for `Css3dObject`s. It integrates with other rendering contexts,
/// such as WebGL, by placing two HtmlElements in front and behind of the Canvas element,
/// allowing the move `Css3dObject`s between these two layers, mimicking z-index ordering.
///
/// To make use of its functionalities, the API user can create a `Css3dSystem` by using
/// the `Css3dRenderer::new_system` method which creates and manages instances of
/// `Css3dObject`s.
```